views/js/ui/maths/calculator/core/board.js

```
@@ -445,20 +455,21 @@ define([
                 // - it is the last result, and the term to add is not an operator
-                if (!tokensHelper.isOperator(term.type) && tokensList.length === 1 && ((currentToken.type === 'NUM0' && name !== 'DOT') || currentToken.type === 'ANS')) {
+                if (!tokensHelper.isOperator(term.type) && !rePrefixedTerm.test(term.value) && tokensList.length === 1 && ((currentToken.type === 'NUM0' && name !== 'DOT') || currentToken.type === 'ANS')) {
                     this.replace(value);
                 } else {
                     // simply add the term, with potentially spaces around
-                    if (!tokensHelper.isSeparator(term.type)) {
+                    if (expression && !tokensHelper.isSeparator(term.type)) {
                         isIdentifier = tokensHelper.isIdentifier(term.type);
+                        needSpace = tokenNeedSpace(currentToken);
 
-                        // prepend space when the either the term to add or the previous term is an identifier
-                        if (position && (tokensHelper.isIdentifier(currentToken) || (isIdentifier && !tokensHelper.isSeparator(currentToken)))) {
+                        // prepend space when either the term to add or the previous term is an identifier
+                        if (position && needSpace) {
                             value = ' ' + value;
                         }
 
-                        // append space when the either the term to add or the previous term is an identifier
-                        if (position < expression.length && (tokensHelper.isIdentifier(nextToken) || (isIdentifier && !tokensHelper.isSeparator(nextToken)))) {
+                        // append space when either the term to add or the next term is an identifier
+                        if ((!position && needSpace) || (position < expression.length && tokenNeedSpace(nextToken))) {
                             value += ' ';
```

I don't mean to nitpick and wouldn't let the PR fail because of this but a variable with the meaning _something needs space_ should be called `needsSpace`, not `needSpace` (3rd person vs 1st person)